rudder-core/src/main/scala/com/normation/rudder/reports/execution/ReportsExecutionService.scala

```diff
@@ -221,11 +222,23 @@ class ReportsExecutionService (
             logger.error("Root exception was: ", ex)
           }
         case Full(x) => //youhou
-          //save compliance in DB
           complianceRepos.saveRunCompliance(x.values.toList)
           logger.trace("Cache for compliance updates after new run received")
       }
     }
+
+    // TODO: save compliance for each run
```

Comment:
Yes, that's the interesting part :) It's not trivial, because it can be heavy, so we will need some queue logic with it's own "done until here" index, or some oracle to know what runs are not calculated (a little more effecient than "select all finish run not in compliance table", I fear)